pkg/apis/v1alpha1/register.go

```diff
@@ -107,6 +107,7 @@ var (
 	LabelInstanceAcceleratorName              = LabelDomain + "/instance-accelerator-name"
 	LabelInstanceAcceleratorManufacturer      = LabelDomain + "/instance-accelerator-manufacturer"
 	LabelInstanceAcceleratorCount             = LabelDomain + "/instance-accelerator-count"
+	AnnotationAWSNodeTemplateHash             = LabelDomain + "/awsnodetemplate-hash"
```

Comment:
can we call it providerRefHash or something similar? Ideally the machine should have this hash, but the label value should be expandable to other provider implementations.
```suggestion
	AnnotationAWSNodeTemplateHash             = LabelDomain + "/awsnodetemplate-hash"
```